extensions/mutiny/runtime/src/main/java/io/quarkus/mutiny/runtime/MutinyInfrastructure.java

```diff
@@ -16,7 +17,9 @@
 
     public static final String VERTX_EVENT_LOOP_THREAD_PREFIX = "vert.x-eventloop-thread-";
 
-    public void configureMutinyInfrastructure(ExecutorService exec) {
+    public void configureMutinyInfrastructure(ExecutorService exec, ShutdownContext shutdownContext) {
+        //mutiny leaks a ScheduledExecutorService if you don't do this
+        Infrastructure.getDefaultWorkerPool().shutdown();
```

Comment:
@cescoffier I think this needs to be addressed in Mutiny. When you call setDefaultExecutor it gets wrapped in a ScheduledExecutor that starts a thread, which then can't really be shutdown without a hack like this.

This thread then ends up holding onto class loaders which is a big problem if you add too many dev mode tests to a module.

Reply:
We can add a method to shutdown the executor. 

Otherwise, I don't see how it can be fixed. Mutiny needs a Scheduled Executor (for several operators). We can delay the creation of the thread until the first call, however, even with this, once used, it will be there. 

CC @jponge

Reply:
You could also shut down the scheduled executor when a new one is being set.

From a Quarkus PoV what I have done here will stop the leaks, it just feels like a hack :-)